processor/groupbytraceprocessor/processor.go

```diff
@@ -116,69 +113,64 @@ func (sp *groupByTraceProcessor) Shutdown(ctx context.Context) error {
 
 func (sp *groupByTraceProcessor) onTraceReceived(batch pdata.Traces) error {
 	for i := 0; i < batch.ResourceSpans().Len(); i++ {
-		if err := sp.processTrace(batch.ResourceSpans().At(i)); err != nil {
+		if err := sp.processResourceSpans(batch.ResourceSpans().At(i)); err != nil {
 			sp.logger.Info("failed to process trace", zap.Error(err))
 		}
 	}
 
 	return nil
 }
 
-func (sp *groupByTraceProcessor) processTrace(trace pdata.ResourceSpans) error {
-	if trace.IsNil() {
+func (sp *groupByTraceProcessor) processResourceSpans(rs pdata.ResourceSpans) error {
+	if rs.IsNil() {
 		// should not happen with the current code
-		return errNilTrace
+		return errNilResourceSpans
 	}
 
-	traceID, err := extractTraceID(trace)
-	if err != nil {
-		return fmt.Errorf("couldn't extract trace ID from trace: %w", err)
-	}
+	for _, batch := range splitByTrace(rs) {
+		traceID := batch.traceID
```

Comment:
Please extract this entire block into a separate `processBatch` (or similar) function.
This will reduce indentation of the code and will improve its overall readability.

Reply:
Done.